lapsharp: blank frame buffer stride region

jstebbins opened the pull request:
Zero is not black.  Stride is used in computations and is assumed black.

Fixes https://github.com/HandBrake/HandBrake/issues/1751

**Test on:**

- [x] Windows 10+  (via MinGW)
- [x] macOS 10.13+
- [x] Ubuntu Linux

jstebbins committed:
lapsharp: blank frame buffer stride region

Zero is not black.  Stride is used in computations and is assumed black.

Fixes https://github.com/HandBrake/HandBrake/issues/1751

jstebbins commented:
Question, would it be better to mirror the final pixels in a line rather than using black?

bradleysepos commented:
@jstebbins I had diagnosed the problem but not made time to solve it. Thanks for fixing this; your solution is better than anything I would have done local to `lapsharp.c`.

Mirroring would be a nice addition here but as long as there are not serious artifacts (shouldn't be), this should suffice as a bug fix for 1.2.1.

I'll be bothering you about more general border mirrors and memory pool stuff soon. NLMeans has similar needs and I was already thinking we could be doing some of this at a lower level than the filters themselves.

bradleysepos commented:
Ship it!

jstebbins committed:
lapsharp: mirror image data into stride region

Adds a function to mirror image data into stride region. Mirroring the
data results in a less visible artefact down the right edge of the
image. Blanking resulted in a dark edge artefact.

Keeps function that blanks stride region and improves it's speed,
although it's not currently used.

jstebbins commented:
I went ahead and added mirroring since I figured it would be useful and this was a good place to test it ;)

bradleysepos commented:
Works, ship it!
